src/operator/operator_common.h

```diff
@@ -309,6 +309,34 @@ inline void ParamParser(nnvm::NodeAttrs* attrs) {
   attrs->parsed = std::move(param);
 }
 
+/*! \brief Callback class to allow for convenient development and testing */
+template<typename Type>
+class Callbacker {
```

Comment:
The reason that there is a callback call is to inspect intermediate values (print them, for instance) without having to include the test_util.h (for example) in the main build tree.  When not being tested, it has no overhead (production code doesn't make callbacks).